functions/Add-DbaAgDatabase.ps1

```diff
@@ -134,98 +134,106 @@ function Add-DbaAgDatabase {
+        } catch {
+            Stop-Function -Message "Failure connecting to secondary instance" -ErrorRecord $_ -Continue
+        }
+    } else {
+        $secondaryInstances = Connect-DbaInstance -SqlInstance $Secondary -SqlCredential $SecondarySqlCredential
+    }
 
-            if ($SeedingMode -ne "Automatic") {
-                if ((Get-DbaDatabase -SqlInstance $ag.Parent -Database $db.Name).LastFullBackup -eq 'Monday, January 1, 0001 12:00:00 AM') {
-                    Stop-Function -Message "Cannot add $($db.Name) to $($ag.Name) on $($ag.Parent.Name). An initial full backup must be created first." -Continue
-                }
-            }
+    if ($SeedingMode -ne "Automatic") {
+        if ((Get-DbaDatabase -SqlInstance $ag.Parent -Database $db.Name).LastFullBackup -eq 'Monday, January 1, 0001 12:00:00 AM') {
+            Stop-Function -Message "Cannot add $($db.Name) to $($ag.Name) on $($ag.Parent.Name). An initial full backup must be created first." -Continue
+        }
+    }
 
-            if ($SeedingMode -eq "Automatic") {
-                # first check
-                if ($Pscmdlet.ShouldProcess($Primary, "Backing up $db to NUL")) {
-                    $null = Backup-DbaDatabase -BackupFileName NUL -SqlInstance $Primary -SqlCredential $SqlCredential -Database $db
-                }
-            }
+    if ($SeedingMode -eq "Automatic") {
+        # first check
+        if ($Pscmdlet.ShouldProcess($Primary, "Backing up $db to NUL")) {
+            $null = Backup-DbaDatabase -BackupFileName NUL -SqlInstance $Primary -SqlCredential $SqlCredential -Database $db
+        }
+    }
 
-            if ($Pscmdlet.ShouldProcess($ag.Parent.Name, "Adding availability group $db to $($db.Parent.Name)")) {
-                try {
-                    $agdb = New-Object Microsoft.SqlServer.Management.Smo.AvailabilityDatabase($ag, $db.Name)
-                    # something is up with .net create(), force a stop
-                    Invoke-Create -Object $agdb
-                    Get-DbaAgDatabase -SqlInstance $ag.Parent -Database $db.Name
-                } catch {
-                    Stop-Function -Message "Failure" -ErrorRecord $_ -Continue
-                }
-            }
+    if ($Pscmdlet.ShouldProcess($ag.Parent.Name, "Adding availability group $db to $($db.Parent.Name)")) {
+        try {
+            $agdb = New-Object Microsoft.SqlServer.Management.Smo.AvailabilityDatabase($ag, $db.Name)
+            # something is up with .net create(), force a stop
+            Invoke-Create -Object $agdb
+            Get-DbaAgDatabase -SqlInstance $ag.Parent -Database $db.Name
+        } catch {
+            Stop-Function -Message "Failure" -ErrorRecord $_ -Continue
+        }
+    }
 
-            foreach ($replica in $secondaryReplicas) {
+    foreach ($instance in $secondaryInstances) {
```

Comment:
$instance is generally what we go for but I think it'd be beneficial to say $secondaryinstance so it's not confused with Primary.

Reply:
Replaced that variable with $secondaryInstance as discussed